### PR TITLE
[SYCL][MATRIX] Adding test cases for testing get_coord() feature.

### PR DESCRIPTION
**Pull request by arnamoy10**
Currently, the tests are marked for failure, they will pass once the ICG
backend support is there.

The header file support is in this PR: https://github.com/intel/llvm/pull/7851

### CODE REVIEW

**Review by dkhaldi — COMMENTED** [review]
Tests look good but the XMX8 versions are missing.

**Comment by arnamoy10** [comment]
> Tests look good but the XMX8 versions are missing.

Added those tests

**Review by dkhaldi — APPROVED** [review]
LGTM

**Review by againull — APPROVED** [review]

File: SYCL/Matrix/get_coord_bf16_matB_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+           //  each WI calculates local sum of cols
+           for (int i = 0; i < wiData.length(); ++i) {
+             // get the index of the element in the submatrix
+             auto dataItem = wiData[i];
+             auto [row, col] = dataItem.get_coord();
+
+             // Calculation of global index
+             int sg_idx = (int)global_idy / SG_SZ;
+             global_index = col + sg_idx * 4 /*VNNI_FACTOR*/ * SG_SZ;
```
Comment (Nuullll) [inline SYCL/Matrix/get_coord_bf16_matB_impl.hpp:166]: Should this line be `global_index = col * 4 + row % 4 + sg_idx * 4 /*VNNI_FACTOR*/ * SG_SZ;` instead?

`[row, col]` are the coordinates within the non-VNNIed subgroup matrix (32x16). I think we need to map `[row, col]` back to VNNIed indices when summing columns.

**Pull request merged**